myapp/data/match.go

```diff
@@ -144,3 +161,24 @@ func (m *Match) Insert(thematch Match) (int, error) {
 
 	return id, nil
 }
+
+func (m *Match) MarkAsViewedForUser(userID int, otherUserID int) error {
```

Comment:
Instead of taking in two users for this, why don't we just take in the match with a boolean telling us which to set to true?